### ISSUE TITLE
Dependency installed via apt-install vs pip?

### ISSUE THREAD

**Issue by Phusionraze** [yt-dlp/yt-dlp#5250]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.10.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I'm running yt-dlp in WSL. I'm not very knowledgeable of how the Linux environment works. Does it make a difference whether a dependency is installed with `sudo apt install` or `pip install`?

As example, brotli can be installed both ways. If installed with pip, I need to enter python to see it, else I'm prompted to install it with sudo apt-install too. Can yt-dlp find either version?

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by flashdagger** [comment]
If you depend on brotli support, you'll need either `brotli` or `brotlicffi` installed as python packages.
The recommendation is to use `brotli` for CPython and `brotlicffi`  everywhere else (PyPy, etc).
see https://github.com/python-hyper/brotlicffi

**Comment by dirkf** [comment]
> Does it make a difference whether a dependency is installed with sudo apt install or pip install

When a Python program imports a module, it should be able to find either type. But the program should be using the same Python that ran _pip_ to install the module or for which the module package from the distro repository was installed.

>...  I need to enter python to see it ...

How are you trying to "see" it?

**Comment by Phusionraze** [comment]
> How are you trying to "see" it?

Sorry for the vague description. I normally determine whether a package is installed in WSL just by entering its name which in the case of `brotli` returns `Command 'brotli' not found, but can be installed` (...). For packages installed with `pip` I assumed I had to enter the name inside a Python session to check installation status, which is what I meant by "see it". In hindsight I see this is however wrong as Python returns `NameError` when I do that. `pip list` works better.

> When a Python program imports a module, it should be able to find either type. But the program should be using the same Python that ran pip to install the module or for which the module package from the distro repository was installed.

Thanks for the info, that makes sense. I started wondering if there's an easy way to check if yt-dlp uses the same Python installation as `pip`, but I guess I guess the easiest solution is simply running `yt-dlp --verbose` to see what installed dependencies it can find.

> The recommendation is to use `brotli` for CPython and `brotlicffi` everywhere else (PyPy, etc).

Thanks for the tip. In my case I currently have no need for other Python versions than the one needed to run yt-dlp, but it's useful to know different versions of packages may cater to different Python versions.
